terraform/monitoring/dashboards/frontend_dashboard.json

```diff
@@ -0,0 +1,153 @@
+{
+  "displayName": "Frontend Service Dashboard",
+  "gridLayout": {
+    "columns": "2",
+    "widgets": [
+      {
+        "title": "CPU limit utilization for stackdriver-sandbox, frontend",
+        "xyChart": {
+          "dataSets": [
+            {
+              "timeSeriesQuery": {
+                "timeSeriesFilter": {
+                  "filter": "metric.type=\"kubernetes.io/container/cpu/limit_utilization\" resource.type=\"k8s_container\" resource.label.\"cluster_name\"=\"stackdriver-sandbox\" metadata.user_labels.\"app\"=\"frontend\"",
+                  "aggregation": {
+                    "perSeriesAligner": "ALIGN_MEAN"
+                  },
+                  "secondaryAggregation": {}
```

Comment:
Did you create these dashboards via the UI, and then download the JSON? There is an unfortunate bug where the UI is writing empty secondaryAggregation objects and empty timeshiftDuration fields.

Since this is an example dashboard I think we should try to keep it as clean as possible. I would recommend removing all empty secondaryAggregation objects and timeshiftDuration set to "0s".

Reply:
Yeah I retrieved them with the API. Good idea - I pruned away all the empty secondaryAggregations and timeShiftDuration="0s". Thanks for catching that.